src/core/utils/power_calibration.hpp

```diff
@@ -0,0 +1,160 @@
+     * @retval kErrorNoBufs       No available entry in the power calibration table.
+     * @retval kErrorInvalidArgs  The @p aChannel, @aActualPower or @p aRawPowerSetting is invalid or the
+     *                            @ aActualPower already exists in the power calibration table;
+     *
+     */
+    Error AddCalibratedPower(uint8_t        aChannel,
+                             int16_t        aActualPower,
+                             const uint8_t *aRawPowerSetting,
+                             uint16_t       aRawPowerSettingLength);
+
+    /**
+     * Clear all calibrated powers from the power calibration table.
+     *
+     */
+    void ClearCalibratedPowers(void);
+
+    /**
+     * Set the target power for the given channel.
+     *
+     * @param[in]  aChannel      The radio channel.
+     * @param[in]  aTargetPower  The target power in 0.01dBm. Passing `INT16_MAX` will disable this channel.
+     *
+     * @retval  kErrorNone         Successfully set the target power.
+     * @retval  kErrorInvalidArgs  The @p aChannel or @p aTargetPower is invalid.
+     *
+     */
+    Error SetChannelTargetPower(uint8_t aChannel, int16_t aTargetPower);
+
+    /**
+     * Get the raw power setting for the given channel.
+     *
+     * Platform radio layer should parse the raw power setting based on the radio layer defined format and set the
+     * parameters of each radio hardware module.
+     *
+     * @param[in]      aChannel                The radio channel.
+     * @param[out]     aRawPowerSetting        A pointer to the raw power setting byte array.
+     * @param[in,out]  aRawPowerSettingLength  On input, a pointer to the size of @p aRawPowerSetting.
+     *                                         On output, a pointer to the length of the raw power setting data.
+     *
+     * @retval  kErrorNone         Successfully got the target power.
+     * @retval  kErrorInvalidArgs  The @p aChannel is invalid or @p aRawPowerSetting is nullptr.
+     * @retval  kErrorNotFound     The raw power setting for the @p aChannel was not found.
+     *
+     */
+    Error GetRawPowerSetting(uint8_t aChannel, uint8_t *aRawPowerSetting, uint16_t *aRawPowerSettingLength);
+
+private:
+    class CalibratedPower
+    {
+    public:
+        static constexpr uint16_t kInvalidPower           = INT16_MAX;
+        static constexpr uint16_t kMaxRawPowerSettingSize = OPENTHREAD_PLATFORM_CONFIG_RAW_POWER_SETTING_SIZE;
+
+        CalibratedPower()
+            : mActualPower(kInvalidPower)
+            , mRawPowerSettingLength(0)
+        {
+        }
+
+        void    Init(int16_t aActualPower, const uint8_t *aRawPowerSetting, uint16_t aRawPowerSettingLength);
+        Error   GetRawPowerSetting(uint8_t *aRawPowerSetting, uint16_t *aRawPowerSettingLength);
+        int16_t GetActualPower(void) const { return mActualPower; }
+
+    private:
+        int16_t  mActualPower;
+        uint8_t  mRawPowerSetting[kMaxRawPowerSettingSize];
```

Comment:
Here also can consider using shorter names (since already under a namespace from the `class` name), maybe `mSettings` and `mLength`?